includes/admin/class-sensei-learner-management.php

```diff
@@ -790,8 +790,20 @@ public function add_new_learners() {
 
 			switch ( $post_type ) {
 				case 'course':
+					$base_query_args        = [
+						'posts_per_page' => -1,
+						'fields' => 'ids'
+					];
+					$learner_manager = Sensei_Learner::instance();
+					$completed_courses   = $learner_manager->get_enrolled_completed_courses_query( $user_id, $base_query_args );
+
 					// Complete each lesson if course is set to be completed.
-					if ( $result && isset( $_POST['add_complete_course'] ) && 'yes' === $_POST['add_complete_course'] ) {
+					if (
+						$result
+						&& isset( $_POST['add_complete_course'] )
+						&& 'yes' === $_POST['add_complete_course']
+						&& ! in_array( $course_id, $completed_courses->posts, true )
```

Comment:
I'm not 100% sure but I think you could simplify the code by using `Sensei_Utils::user_completed_course` instead of using the learner instance.

Reply:
Thanks @m1r0 - that is much easier :) 
I've updated the code now!

Reply:
Nice! Could you check the [linter errors](https://github.com/Automattic/sensei/runs/7678821873?check_suite_focus=true) as well?